### PR TITLE
cleanup: delete tempDir correctly after flexvloume_test is executed

### PR DESCRIPTION
**Pull request by njuptlzf**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
flake test
The `watcher `sometimes recreates the `tmpDir `because `defer os.RemoveAll(tmpDir)` deletes the tmpDir, causing redundancy.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/volume/flexvolume/flexvolume_test.go [hunk 1]
```diff
@@ -123,7 +124,9 @@ func TestCanSupport(t *testing.T) {
 	plugMgr := volume.VolumePluginMgr{}
 	runner := exec.New()
 	installPluginUnderTest(t, "kubernetes.io", "fakeAttacher", tmpDir, execScriptTempl1, nil)
-	plugMgr.InitPlugins(nil, GetDynamicPluginProber(tmpDir, runner), volumetest.NewFakeVolumeHost(t, "fake", nil, nil))
+	if err := plugMgr.InitPlugins(nil, GetDynamicPluginProberWithoutWatcher(tmpDir, runner), volumetest.NewFakeVolumeHost(t, "fake", nil, nil)); err != nil {
```
Comment (msau42) [inline pkg/volume/flexvolume/flexvolume_test.go:127]: Can you explain how this fixes the tempdir cleanup issue? It's not obvious to me.

Comment (njuptlzf) [reply pkg/volume/flexvolume/flexvolume_test.go:127]: The` GetDynamicPluginProber() `method starts the real watcher. When `defer os.RemoveAll(tmpDir) `is executed, tmpdir is recreated. I think the real watcher is redundant.

**Pull request merged**